packages/yarnpkg-pnp/sources/types.ts

```diff
@@ -90,6 +90,9 @@ export type PnpSettings = {
   // getDependencyTreeRoots function. They are typically the workspace
   // locators.
   dependencyTreeRoots: Array<PhysicalPackageLocator>,
+
+  // node_modules nohoist pattern list for each workspace (used in PnP loose mode)
+  nmNohoistPatterns: Array<{relativeCwd: string, nohoistPatterns: Array<string>}>,
```

Comment:
I don't think it's a good idea to leak `nohoist`, which is just a NM-specific temporary solution (in the best case scenario, but let's be real, nothing is more permanent than a temporary solution), into the PnP API. In the best case scenario, I think that `nmNohoistPatterns` should be passed directly to the `Installer`s that need it without going through the `PnpSettings` of the PnP API, while another alternative could be having an `extras: Record<string, string>` field on the `PnpSettings` interface. 🤔